rs/replicated_state/src/canister_state/queues.rs

```diff
@@ -174,78 +124,69 @@ pub struct NewCanisterQueues {
 pub struct CanisterOutputQueuesIterator<'a> {
     /// Priority queue of non-empty output queues. The next message to be popped
     /// / peeked is the one at the head of the first queue.
-    queues: VecDeque<(&'a CanisterId, &'a mut OutputQueue)>,
-
-    /// Number of messages that can be popped before the iterator finishes.
-    size: usize,
+    queues: VecDeque<(&'a CanisterId, &'a mut CanisterQueue)>,
 
-    /// The canister's memory usage stats, to be updated as messages are popped.
-    memory_stats: &'a mut MemoryUsageStats,
+    pool: &'a mut MessagePool,
 
-    /// Canister output queue stats, to be updated as messages are popped.
-    queue_stats: &'a mut OutputQueuesStats,
```

Comment:
Why do we no longer need to queue stats in the iterator? It would seem that popping a message has an effect on the stats, no?

Reply:
`OutputQueueStats` held the number of messages in output queues (now maintained by `message_pool::MessageStats`) and the amount of cycles in output queues (which was no longer used anywhere, except in a failed attempt to ensure that no cycles were fabricated in any given execution round).

So I dropped `OutputQueueStats` (and `InputQueueStats`, for similar reasons) altogether. All the stats are now provided by the combination of `message_pool::MessageStats` (anything to do with the number or size of messages) and `queues::QueueStats` (memory and queue slot reservations; plus the `transient_stream_responses_size_bytes`, populated from the streams every round).

Reply:
I see. I had somehow assumed that one would have to adjust the byte taken by responses when popping a response from the queue. Is this done somewhere else, or is this ignored here?

Reply:
The `MessagePool` updates `MessageStats` whenever a message is taken out of the pool (whether it's popped, expired or shed).